docs/src/references/agent/device-profile.md

```diff
@@ -0,0 +1,270 @@
+  "name": "prod-profile",
+  "version: "v1"
+}'
+```
+
+If the current profile information is not known upfront, publish an empty JSON (`{}`) instead.
+The firmware information, supported software and configuration types must be declared separately with their respective capability messages.
+
+# Device profile command
+
+Once the `device_profile` capability is declared, the device can receive `device_profile` commands
+by subscribing to `<root>/<device-topic-id>/cmd/device_profile/+` MQTT topics.
+For example, subscribe to the following topic for the `main` device:
+
+```sh te2mqtt formats=v1
+tedge mqtt sub 'te/device/main///cmd/device_profile/+'
+```
+
+A `device_profile` command with `id` "1234" is triggered as follows:
+
+```sh te2mqtt formats=v1
+tedge mqtt pub -r 'te/device/main///cmd/device_profile/1234' '{
+  "status": "init",
+  "name": "prod-profile",
+  "version": "v2",
+  "profile": [
+    {
+      "operation": "firmware_update",
+      "skip": false,
+      "payload": {
+        "name": "core-image-tedge-rauc",
+        "remoteUrl": "https://abc.com/some/firmware/url",
+        "version": "20240430.1139"
+      }
+    },
+    {
+      "operation": "software_update",
+      "skip": false,
+      "payload": {
+        "updateList": [
+          {
+            "type": "apt",
+            "modules": [
+              {
+                "name": "c8y-command-plugin",
+                "version": "latest",
+                "action": "install"
+              },
+              {
+                "name": "collectd",
+                "version": "latest",
+                "type": "apt",
+                "action": "install"
+              }
+            ]
+          }
+        ]
+      }
+    },
+    {
+      "operation": "config_update",
+      "skip": false,
+      "payload": {
+        "type": "collectd.conf",
+        "remoteUrl":"https://abc.com/some/collectd/conf",
+        "path": "/etc/collectd/collectd.conf"
+      }
+    },
+    {
+      "operation": "software_update",
+      "skip": false,
+      "payload": {
+        "updateList": [
+          {
+            "type": "apt",
+            "modules": [
+              {
+                "name": "jq",
+                "version": "latest",
+                "action": "install"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  ]
+}'
+```
+
+The profile definition in the payload is an array of operations.
+Each operation could be a `firmware_update`, `software_update` or `config_update`.
+The operations are executed in the order in which they are defined in the profile definition, by default.
+There is no restriction on the order of modules in a profile and hence can be defined in any preferred order.
+For example, additional software can be installed or configurations updated before the firmware is updated.
+This default execution order can also be overridden in the workflow definition, by updating the order in the `scheduled` state.
```

Comment:
Do we need to restrict this to a specific state? Can't we just assume that the sorting has happened prior to starting to process the items?

Reply:
Yeah, that's true. It just has to be any state prior to the `executing` state. I was just trying to keep the example as minimal as possible reusing the existing states as much as possible without introducing any additional non-essential states, as that might confuse the user that those states are also mandatory for device profiles.

But yeah, I'll rephrase the sentence to clarify that the sorting can be done in this state or any other subsequent state before `executing`.

Reply:
Rephrased the sentence as suggested.